db/sql/003-function-get-impacted-accruals.sql

```diff
@@ -0,0 +1,55 @@
+-- =============================================================================
+-- Description: Returns list of Accruals impacted by Time Entry. Covering dates:
+--              FROM the earliest impacted accrual or Time Entry start date, whichever is earliest,
+--              minus one day ('day before' used in calculation of cumulative totals)
+--              UNTIL the end of agreement corresponding to the latest impacted accrual or Time
+--              Entry end date, whichever is latest.
+--
+-- Parameters:
+-- @p_tenant_id             - Tenant ID
+-- @p_person_id             - Person ID generated by Person container
+-- @p_time_entry_id         - Time Entry ID generated by Timecard container
+-- @p_time_entry_start_date - Time Entry start date (it can be before the first found contribution)
+-- @p_time_entry_end_date   - Time Entry end date (it can be after the last found contribution)
+--
+-- Returns: Set of Accrual table records meeting the input criteria
+-- =============================================================================
+
+CREATE OR REPLACE FUNCTION accruals.get_impacted_accruals(
```

Comment:
function looks good, I'm thinking we may want to thing about what indexes we have. One that jumped out to me was indexing start and end dates on agreements, along with the tenant and person ids

Reply:
Would you want to see a change set for this as part of this ticket, or is this just a note to keep in mind moving forward?

Reply:
I'd make it part of this ticket but I'd have to take a look to check that it useful to add the index. It's only because I suspect that the start and end dates are not part of the primary key  and that we've not added any additional indexes.

Reply:
I've had a look and it looks like it would be too early an optimisation at this point. At the moment with such little data present it's just performing a seq scan and pushing more data in wouldn't represent what reality may look like in the future.